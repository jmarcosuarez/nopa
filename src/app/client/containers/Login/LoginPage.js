import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { compose, curry, pick, prop } from 'ramda';
import { validate } from 'spected';
import { Layout, LoginForm } from '../../components';
import revalidation from './Revalidation';
import styles from './Login.css';

const isNotEmpty = a => a.trim().length > 0;
const hasCapitalLetter = a => /[A-Z]/.test(a);
const isGreaterThan = curry((len, a) => (a > len));
const isLengthGreaterThan = len => compose(isGreaterThan(len), prop('length'));

const basicValidationRules = {
  surname: [
    [isNotEmpty, 'Surname should not be empty.'],
  ],
  memorableWord: [
    [isLengthGreaterThan(7), 'Minimum MemorableWord length of 8 is required.'],
    [hasCapitalLetter, 'MemorableWord should contain at least one uppercase letter.'],
  ],
};

const validateFn = validate(() => [], a => a);
const SimpleForm = revalidation(
  {
    validate: {
      all: data => 
        validateFn(basicValidationRules, data),
      input: (name, value) => 
        validateFn(pick([name], basicValidationRules), { [name]: value }),
    },
  },
)(LoginForm);

export default class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      form: {
        surname: '', 
        sortCode: '23-43-54',
        accountNumber: '09434534343434',
        passCode: '0943',
        memorableWord: '',
      } };
  }

  render() {
    return (
      <Layout>
        <div className={styles.main}>
          <h1>Which bank does this account belong to?</h1>
          <p>Track all of your payments by connecting as many bank accounts as you&#39;d<br />
              like to your Nopa account and get updates on your balance instantly. Plus it&#39;s free.</p>   
          <SimpleForm
            values={this.state.form}
            submitCb={this.props.onLogInUser}
          /> 
        </div>
      </Layout>
    );
  }
}

LoginPage.propTypes = {
  onLogInUser: PropTypes.func,
};

LoginPage.defaultProps = {
  onLogInUser: () => {},
};