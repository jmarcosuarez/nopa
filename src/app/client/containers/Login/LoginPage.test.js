/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoginPage from './LoginPage';

describe('Renders correctly', () => {
  it('should render correctly', () => {
    const output = shallow(
      <LoginPage />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});