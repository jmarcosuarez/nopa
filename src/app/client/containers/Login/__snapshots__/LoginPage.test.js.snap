// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section>
  <div
    className="header"
  >
    <div
      className="logo"
    />
    <div
      className="signin"
    >
      <div
        className=""
      >
        <a
          className="button"
          onClick={[Function]}
          style={Object {}}
        >
          Log In
        </a>
      </div>
    </div>
  </div>
  <div
    className="layout"
  >
    <div
      className="main-content"
    >
      <h1>
        Which bank does this account belong to?
      </h1>
      <p>
        Track all of your payments by connecting as many bank accounts as you'd
        <br />
        like to your Nopa account and get updates on your balance instantly. Plus it's free.
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-input"
        >
          <label
            htmlFor="surname"
          >
            Surname
          </label>
          <input
            data-error={undefined}
            id="surname"
            name="surname"
            onChange={[Function]}
            placeholder="Surname"
            type={undefined}
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="sortCode"
          >
            Sort code
          </label>
          <input
            data-error={undefined}
            id="sortCode"
            name="sortCode"
            onChange={[Function]}
            placeholder="Sort code"
            type={undefined}
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="accountNumber"
          >
            Account number
          </label>
          <input
            data-error={undefined}
            id="accountNumber"
            name="accountNumber"
            onChange={[Function]}
            placeholder="Account number"
            type={undefined}
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="passCode"
          >
            Passode
          </label>
          <input
            data-error={undefined}
            id="passCode"
            name="passCode"
            onChange={[Function]}
            placeholder="Passode"
            type={undefined}
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="memorableWord"
          >
            Memorable word
          </label>
          <input
            data-error={undefined}
            id="memorableWord"
            name="memorableWord"
            onChange={[Function]}
            placeholder="Memorable word"
            type={undefined}
          />
          <div
            className="errors"
          />
        </div>
        <input
          className="button"
          type="submit"
          value="Login & connect"
        />
      </form>
    </div>
  </div>
  <section
    className="partners"
  >
    <div>
      <span>
        Our partners:
      </span>
    </div>
    <div>
      <div>
        <img
          alt="Airbnb"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Metro"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Pariti"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
      <div>
        <img
          alt="Unshackled"
          className="Nopa"
          src="test-file-stub"
        />
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div>
      <p>
        © Nopa Limited 2017 All rights reserved. ‘Nopa’ and the Nopa logo are trade marks of Nopa Limited. Nopa is a member of CIFAS - the UK's leading anti-fraud association, and we are registered with the Office of the Information Commissioner (No. Z879078).
      </p>
      <p>
        Nopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG. Nopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register under firm registration number 563134.
      </p>
    </div>
  </footer>
</section>
`;
